plugins/SpectrumAnalyzer/SaProcessor.h

```diff
@@ -42,19 +45,35 @@ class SaProcessor
 	explicit SaProcessor(SaControls *controls);
 	virtual ~SaProcessor();
 
-	void analyse(sampleFrame *in_buffer, const fpp_t frame_count);
+	// analysis thread and a method to terminate it
+	void analyze(ringbuffer_t<sampleFrame> &ring_buffer, QWaitCondition &notifier);
+	void terminate() {m_terminate = true;}
 
 	// inform processor if any processing is actually required
 	void setSpectrumActive(bool active);
 	void setWaterfallActive(bool active);
+	void flipRequest() {m_flipRequest = true;}	// request refresh of history buffer
 
 	// configuration is taken from models in SaControls; some changes require
 	// an exlicit update request (reallocation and window rebuild)
 	void reallocateBuffers();
 	void rebuildWindow();
 	void clear();
+	void clearHistory();
+
+	const float *getSpectrumL() const {return m_normSpectrumL.data();}
+	const float *getSpectrumR() const {return m_normSpectrumR.data();}
+	const uchar *getHistory() const {return m_history.data();}
 
 	// information about results and unit conversion helpers
+	const unsigned int &inBlockSize() const {return m_inBlockSize;}
```

Comment:
I wanted to make sure the value won't be copied, since `inBlockSize()` could be called relatively often. Previously it just read an existing value, now it calls a function that would have to copy the value to be returned every time, right?

Reply:
Hmm, you are probably right. I don't use C that often so know all the details about what and when is / can be optimized, so I mostly assume the worst when I'm not sure. But if you think it makes no difference, I'm happy to get rid of the reference, it definitely does not help readability.

Reply:
If you're trying to understand what is equivalent to what, then I'd recommend https://godbolt.org/ as a testbed. It makes it simpler to see how different forms of code compile and it shows the impact of various compiler flags (e.g. optimization).